dot/core/service.go

```diff
@@ -53,7 +55,8 @@ type Service struct {
 	codeSubstitutedState CodeSubstitutedState
 
 	// Keystore
-	keys *keystore.GlobalKeystore
+	keys        *keystore.GlobalKeystore
+	newInstance newWasmerInstanceFunc
```

Comment:
why did you add this as a `Service` attribute? It's not used anywhere else I believe, so I think it should be localized to `handleCodeSubstitution` as a parameter.

Reply:
I think we should change it back. I added a comment in the resolved conversation. https://github.com/ChainSafe/gossamer/pull/2239#discussion_r810232906